src/coreclr/src/vm/eventpipe.cpp

```diff
@@ -435,6 +450,27 @@ void EventPipe::Disable(EventPipeSessionID id)
     }
     CONTRACTL_END;
 
+    // EventPipe::Disable is called synchronously since the diagnostics server is
```

Comment:
This is fixing a separate pre-existing bug yes? I think its fine to include it for expediency sake but I recommend updating the issue/final commit message to call it out separately.

Reply:
Updated the PR text